crates/iceberg/src/spec/values.rs

```diff
@@ -2714,23 +2660,27 @@ mod tests {
         assert_eq!(parsed_json_value, raw_json_value);
     }
 
-    fn check_avro_bytes_serde(input: Vec<u8>, expected_literal: Literal, expected_type: &Type) {
+    fn check_avro_bytes_serde(
+        input: Vec<u8>,
+        expected_datum: Datum,
+        expected_type: &PrimitiveType,
+    ) {
         let raw_schema = r#""bytes""#;
         let schema = apache_avro::Schema::parse_str(raw_schema).unwrap();
 
         let bytes = ByteBuf::from(input);
-        let literal = Literal::try_from_bytes(&bytes, expected_type).unwrap();
-        assert_eq!(literal, expected_literal);
+        let datum = Datum::try_from_bytes(&bytes, expected_type.clone()).unwrap();
+        assert_eq!(datum, expected_datum);
```

Comment:
This looks quite suspicious.  `Datum::try_from_bytes ` links to <https://iceberg.apache.org/spec/#binary-single-value-serialization>, which is not Avro encoding

Reply:
Hmmm, ok. It has a reason. Strange still

https://github.com/apache/iceberg-rust/pull/20/files#r1288038661

Reply:
[The binary serialization](https://iceberg.apache.org/spec/#binary-single-value-serialization,) is different from [avro encoding](https://iceberg.apache.org/spec/#avro).

Reply:
Yes. I mean doing `check_avro_` with binary serialization is strange..